### PR TITLE
Optionally keep name section in shrink and optimize

### PR DESCRIPTION
**Pull request by kentosugama**
Add an option under the `shrink` command to let the user keep the name sections in the wasm module.
This is useful for debugging and profiling use cases.

```
ic-wasm -o example.wasm example.wasm shrink --optimize O3 --keep-name-section
```

### CODE REVIEW

File: tests/deployable.ic-repl.sh [hunk 1]
```diff
@@ -93,26 +93,33 @@ let S = motoko(file("ok/motoko-gc-instrument.wasm"));
...
 
 classes(file("ok/classes-shrink.wasm"));
+classes(file("ok/classes-shrink-names.wasm"));
+classes(file("ok/classes-optimize.wasm"));
+classes(file("ok/classes-optimize-names.wasm"));
 classes_limit(file("ok/classes-limit.wasm"));
 classes_redirect(file("ok/classes-redirect.wasm"));
 classes(file("ok/classes-nop-redirect.wasm"));
-
-motoko(file("ok/motoko-optimize.wasm"));
```
Comment (chenyan-dfinity) [inline tests/deployable.ic-repl.sh:115]: We don't need to add the shrink and optimize wasms above. Just one test here is enough. It's too much noise to test. Every time we bump the wasm-opt version, we have to update these wasm files.

File: src/bin/main.rs [hunk 2]
```diff
@@ -52,6 +52,8 @@ enum SubCommand {
     Shrink {
         #[clap(long, value_parser = ["O0", "O1", "O2", "O3", "O4", "Os", "Oz"])]
         optimize: Option<String>,
+        #[clap(short, long)]
+        keep_name_section: bool,
```
Comment (chenyan-dfinity) [inline src/bin/main.rs:56]: Why is this a breaking change. If `keep_name_section` is omitted, it gets false, which is the old behavior?

Comment (kentosugama) [reply src/bin/main.rs:56]: When I use the current `main` and run
```
ic-wasm -o motoko-shrink.wasm motoko.wasm shrink
wasm2wat motoko-shrink.wasm -o motoko-shrink.wat
```
it seems that the name section is kept. 
https://github.com/dfinity/ic-wasm/blob/3246f6d4e888064a3895e2cd88853727933c187d/src/bin/main.rs#L65

If my understanding is correct, the current `shrink` keeps the name sections (and drops the metadata sections). Unless this is a bug.

Comment (kentosugama) [reply src/bin/main.rs:56]: Hmm that's strange, the name section should be dropped here:
https://github.com/dfinity/ic-wasm/blob/3246f6d4e888064a3895e2cd88853727933c187d/src/shrink.rs#L20

Comment (chenyan-dfinity) [reply src/bin/main.rs:56]: Name section is special in `walrus` and doesn't appear in the section part. It's controlled by the parser: https://docs.rs/walrus/0.19.0/walrus/struct.ModuleConfig.html#method.generate_name_section

**Review by chenyan-dfinity — APPROVED** [review]
LGTM

**Pull request merged**